pkg/operator/controller/status/controller.go

```diff
@@ -167,6 +197,33 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 			Name:     state.CanaryNamespace.Name,
 		})
 	}
+	if r.config.GatewayAPIEnabled {
+		related = append(related, configv1.ObjectReference{
+			Group:    gatewayapiv1.GroupName,
+			Resource: "gatewayclasses",
+		})
+		if state.haveOSSMSubscription {
+			subscriptionName := operatorcontroller.ServiceMeshOperatorSubscriptionName()
+			related = append(related, configv1.ObjectReference{
+				Group:     operatorsv1alpha1.GroupName,
+				Resource:  "subscriptions",
+				Namespace: subscriptionName.Namespace,
+				Name:      subscriptionName.Name,
+			})
+			if state.IngressNamespace != nil {
+				related = append(related, configv1.ObjectReference{
+					Group:     sailv1.GroupVersion.Group,
+					Resource:  "istios",
+					Namespace: state.IngressNamespace.Name,
+				})
+				related = append(related, configv1.ObjectReference{
+					Group:     gatewayapiv1.GroupName,
+					Resource:  "gateways",
```

Comment:
If we don't have our own HTTPRoutes, GRPCRoutes, or ReferenceGrants, we also don't have our own Gateways.  Gateways are also user workload objects, aren't they?

Reply:
The delineation here is a little blurry.  Gateways are akin to ingresscontrollers, in that the cluster-admin creates them to configure infrastructure for workload, so there is some overlap work workload.  I think it's appropriate for must-gather to include gateways because they could be useful in diagnosing customer issues, and they shouldn't have anything too sensitive (for example, the certificate is in a secret that the gateway references, not in the gateway itself).